src/lib.rs

```diff
@@ -1,5 +1,6 @@
 use metric::{HostMetric, MetricValue};
 use std::{
+    collections::VecDeque,
```

Comment:
memo:
> A double-ended queue implemented with a growable ring buffer.
The "default" usage of this type as a queue is to use push_back to add to the queue, and pop_front to remove from the queue. extend and append push onto the back in this manner, and iterating over VecDeque goes front to back.